### ISSUE TITLE
how to embed a ktrain model under a KerasRegressor call?

### ISSUE THREAD

**Issue by randomgambit** [amaiya/ktrain#381]
Hi @amaiya !

Just a quick question if you have some time. I would like to take advantage of the cross-validation features in scikit and given that `ktrain` is a wrapper around `keras` I though I would be able to wrap a `ktrain` call under a `KerasRegressor` call.

Unfortunately, the following does not seem to work correctly. I tried to wrap your simple text classification example but I get an error. What is the correct way to wrap `ktrain` then?

```
import ktrain
from ktrain import text
import keras
from keras.models import Sequential
from keras.layers import Dense
from tensorflow.keras.wrappers.scikit_learn import KerasRegressor
from sklearn.model_selection import RandomizedSearchCV

def build_model(maxlen = 500, class_names = [] ):

    MODEL_NAME = 'distilbert-base-uncased'
    t = text.Transformer(MODEL_NAME, maxlen, class_names)
    trn = t.preprocess_train(x_train, y_train)
    val = t.preprocess_test(x_test, y_test)
    model = t.get_classifier()
    return model


categories = ['alt.atheism', 'soc.religion.christian','comp.graphics', 'sci.med']
from sklearn.datasets import fetch_20newsgroups
train_b = fetch_20newsgroups(subset='train', categories=categories, shuffle=True)
test_b = fetch_20newsgroups(subset='test',categories=categories, shuffle=True)
(x_train, y_train) = (train_b.data, train_b.target)
(x_test, y_test) = (test_b.data, test_b.target)

#create the keras layer that will allow me to use cross validation, predict(), etc
keras_reg = KerasRegressor(build_model)

param_distribs = {
"maxlen": [100, 500],
"class_names": train_b.target_names
}

rnd_search_cv = RandomizedSearchCV(keras_reg, param_distribs, n_iter=10, cv=3)
rnd_search_cv.fit(x_train, y_train, epochs=100)

```

Thanks!

**Comment by amaiya** [comment]
Perhaps try something like this:

```python
def transformer(MODEL_NAME):
    preproc  = text.Transformer(MODEL_NAME, maxlen=200)
    x_test=df_test['text'].values
    x_test = preproc.preprocess_test(x_test)
    predictions,rmses=[],[]
    data = df_train[['text', 'target']]
    for train_index, val_index in KFold(5).split(data):
        train,val=data.iloc[train_index],data.iloc[val_index]
        x_train=train.excerpt.values
        x_val=val.excerpt.values

        y_train=train.target.values
        y_val=val.target.values

        trn = preproc.preprocess_train(x_train, y_train)
        val = preproc.preprocess_test(x_val, y_val)
        model = preproc.get_regression_model()
        learner = ktrain.get_learner(model, train_data=trn, val_data=val, batch_size=24)
        learner.fit_onecycle(1e-5, 5)
        print('mse',learner.validate())
        pred=learner.predict(val)
        rmse=mean_squared_error(y_val,pred,squared=False)
        print('rmse',rmse)
        predictions.append(learner.predict(x_test))
        rmses.append(rmse)
    return predictions,rmses
```

**Comment by randomgambit** [comment]
thanks @amaiya ! but this somehow embeds the training AND kfold validation in the function. My hope was to create a keras regressor object that will have a fit(), score() methods etc. In particular this will allow the user to call RandomizedSearchCV with the ktrain model but also with other scikit or keras models. That should be possible conceptually given that ktrain is a wrapper around keras... what do you think?

**Comment by amaiya** [comment]
It's true that ktrain is a wrapper to tf.keras, but it doesn't expose the exact same API as Keras , which will probably cause problems for KerasRegressor.    It's still possible, though, if you modify or subclass KerasRegressor with some slight modifications, but I haven't tried it.

**Comment by randomgambit** [comment]
interesting, thanks. I think that would be a nice feature to have. You would leverage all of scikit's nice features.
In any case, keep up the good work!

Thanks
